matchzoo/models/dssm_model.py

```diff
@@ -0,0 +1,88 @@
+"""An implementation of DSSM, Deep Structured Semantic Model."""
+from matchzoo import engine
+from keras.models import Model
+from keras.layers import Dense, Input, Dot
+
+
+class DssmModel(engine.BaseModel):
+    """
+    Deep structured semantic model.
+
+    Examples:
+        >>> model = DssmModel()
+        >>> model.params['lr'] = 0.2
+        >>> model.guess_and_fill_missing_params()
+        >>> if model.all_params_filled():
+        ...     model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params.add(engine.Param('w_initializer', 'glorot_normal'))
+        params.add(engine.Param('b_initializer', 'zeros'))
+        # TODO GET TRI-LETTER DIMENSIONALITY FROM FIT-TRANSFORM AS INPUT SHAPE
+        params.add(engine.Param('input_shapes', [(30000,), (30000,)]))
+        params.add(engine.Param('dim_fan_out', 128))
+        params.add(engine.Param('dim_hidden', 300))
+        params.add(engine.Param('activation_hidden', 'tanh'))
+        params.add(engine.Param('activation_prediction', 'softmax'))
+        params.add(engine.Param('lr', 0.01))
+        params.add(engine.Param('decay', 1e-6))
+        params.add(engine.Param('batch_size', 1024))
+        params.add(engine.Param('num_epochs', 20))
```

Comment:
`batch_size` and `num_epochs` are not part of model structures. We should set them manually for now and add them into a higher level API to models (e.g. something called a `Trainer`) later.